src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -1018,12 +1030,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Get
         End Property
 
-        Private ReadOnly Property IConversionExpression_IsExplicit As Boolean Implements IConversionExpression.IsExplicit
+        Private ReadOnly Property IConversionExpression_ExplicitCastInSource As Boolean Implements IConversionExpression.ExplicitCastInSource
             Get
                 Return Me.ExplicitCastInCode
             End Get
         End Property
 
+        Private ReadOnly Property IConversionExpression_Conversion As IConversion Implements IConversionExpression.Conversion
+            Get
+                Return New Conversion(Me.ConversionKind, method:=Nothing)
```

Comment:
It is very simple all new code (including implementation of new APIs) should be tested and have correct implementation. Spreading bugs around is not an option. The fact that other feature that you happened to base the code off of has zero tests and has incorrect implementation is not an excuse. Also, in order to merge any code into compilers you need a green light from two members of the compiler team. You won't get this sign-off without adequate testing that should involve all possible kinds of conversions as well as making sure that the Conversion instances that you produce are exactly equivalent to those produced by SemanticModel for the same scenarios.